### PR TITLE
    List bugs targeted to a different release than the one being reported on

### PR DESCRIPTION
**Pull request by bparees**
List test name matching bugs that are targeted to the same release as the report as linked.
List test name matching bugs that are not targeted to the same release as the report as associated

fixes https://github.com/openshift/sippy/issues/128

### CODE REVIEW

**Comment by bparees** [comment]
sample output:

image


image

File: pkg/apis/sippy/v1/types.go [hunk 1]
```diff
@@ -50,10 +50,11 @@ type MinimumPassRatesByComponent struct {
...
 type FailingTestBug struct {
-	Name      string              `json:"name"`
-	Url       string              `json:"url"`
-	PassRates map[string]PassRate `json:"passRates"`
-	Bugs      []bugsv1.Bug        `json:"bugs,omitempty"`
+	Name           string              `json:"name"`
+	Url            string              `json:"url"`
+	PassRates      map[string]PassRate `json:"passRates"`
+	Bugs           []bugsv1.Bug        `json:"bugs,omitempty"`
+	AssociatedBugs []bugsv1.Bug        `json:"associatedBugs,omitempty"`
```
Comment (deads2k) [inline pkg/apis/sippy/v1/types.go:57]: The distinction between `bugs` and `associatedBugs` isn't immediately obvious at an API level.  `nonCausalBugs`?

File: pkg/buganalysis/cache.go [hunk 2]
```diff
@@ -21,7 +21,7 @@ import (
 // It is stateful though, so for a time after clearing the data will not be up to date until the Update is called
 type BugCache interface {
 	ListJobBlockingBugs(job string) []bugsv1.Bug
-	ListBugs(release, variant, testName string) []bugsv1.Bug
+	ListBugs(release, variant, testName string, invertReleaseQuery bool) []bugsv1.Bug
```
Comment (deads2k) [inline pkg/buganalysis/cache.go:24]: Prefer a second method that can call a private helper with this bool.  That keeps code paths consistent and makes intent in the calling code very clear and searchable.

**Comment by deads2k** [comment]
a couple minor comments.

File: pkg/apis/sippy/v1/types.go (hunk 1 above)
Comment (bparees) [reply pkg/apis/sippy/v1/types.go:57]: i dunno that noncausalbugs is going to be any more obvious.  would you settle for me adding a comment that explains the distinction?

File: pkg/buganalysis/cache.go (hunk 2 above)
Comment (bparees) [reply pkg/buganalysis/cache.go]: updated

**Pull request merged**